lib/Analysis/MemoryBuiltins.cpp

```diff
@@ -73,7 +73,12 @@ static const std::pair<LibFunc::Func, AllocFnsTy> AllocationFnData[] = {
   {LibFunc::realloc,             {ReallocLike, 2, 1,  -1}},
   {LibFunc::reallocf,            {ReallocLike, 2, 1,  -1}},
   {LibFunc::strdup,              {StrDupLike,  1, -1, -1}},
-  {LibFunc::strndup,             {StrDupLike,  2, 1,  -1}}
+  {LibFunc::strndup,             {StrDupLike,  2, 1,  -1}},
+
+  {LibFunc::rust_allocate,       {MallocLike,  2, 0,  -1}},
+  {LibFunc::rust_allocate_zeroed, {MallocLike,  2, 0,  -1}},
+  {LibFunc::rust_reallocate,     {ReallocLike,  4, 2,  -1}},
+  {LibFunc::rust_reallocate_inplace, {ReallocLike,  4, 2,  -1}}
```

Comment:
Is `rust_reallocate_inplace` really `ReallocLike`? `realloc` invalidates the old pointer on success (even if the block does not move), and `rust_reallocate_inplace` explicitly does not do that.